### PR TITLE
Tutorial Validation Extracts Rules from Markdown

### PR DESCRIPTION
**Pull request by Jxwoon**
Tutorial Code validation now extracts rules from a markdown file in the format:

```tutorialValidationRules
{"rule1":true, "rule2":false}
```
"rule1" corresponds to the name of the rule, and the boolean after the colon indicates if the rule is turned on/off. Associated with each rule is a boolean status flag that indicates if the user passes this rule and a rule message.

### CODE REVIEW

**Review by livcheerful — APPROVED** [review]

File: pxtlib/tutorial.ts [hunk 1]
```diff
@@ -282,6 +290,23 @@ ${code}
         return { header, hint };
     }
 
+    /* Parse Tutorial Validation Rules */
+    function parseTutorialValidationRules(body: string, tutorialCodeValidation?: boolean): pxt.Map<boolean> {
```
Comment (shakao) [inline pxtlib/tutorial.ts:294]: (late comment/this can be in a separate PR) i think this should use JSON.parse instead of manually parsing (see https://github.com/microsoft/pxt/blob/master/pxtlib/util.ts#L276)
